hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/ChecksumFileSystem.java

```diff
@@ -670,6 +670,27 @@ public boolean rename(Path src, Path dst) throws IOException {
     }
   }
 
+  /**
+   * Rename files/dirs.
+   */
+  @Override
+  @SuppressWarnings("deprecation")
+  public void rename(final Path src,
+      Path dst,
+      final Options.Rename... options)
+      throws IOException {
+      fs.rename(src, dst, options);
+
+      Path srcCheckFile = getChecksumFile(src);
+      Path dstCheckFile = getChecksumFile(dst);
+      if (fs.exists(srcCheckFile)) { //try to rename checksum
+        fs.rename(srcCheckFile, dstCheckFile, Options.Rename.OVERWRITE);
```

Comment:
Should this be `options` with `Options.Rename.OVERWRITE` appended? I think all the Rename options work alone, but if we are passing an unbounded list of options, we should probably use them for future proofing.

Reply:
interesting thought. The only current option is trash; don't know if we should retain that. I will add a comment in the javadocs of Options.Rename on this topic

Reply:
+added comments in Option as well as Checksum FS